### ISSUE TITLE
linsolve([ ], x) should return [ ]

### ISSUE THREAD

**Issue by leosartaj** [sympy/sympy#9953]
Currently 

```
>>> from sympy.solvers.solveset import linsolve
>>> linsolve([], x)
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-10-2d1284394b5a> in <module>()
----> 1 linsolve([], x)

/home/leosartaj/projects/sympy/sympy/solvers/solveset.py in linsolve(system, *symbols)
   1224 
   1225         # 3). List of equations Form
-> 1226         if not system[0].is_Matrix:
   1227             A, b = linear_eq_to_matrix(system, symbols)
   1228 

IndexError: list index out of range
```

This should return an empty list and should not fail.

Ping @aktech

**Comment by aktech** [comment]
ping @Devyani-Divs

**Comment by leosartaj** [comment]
As decided during the open-space at PyCon India, it will better to return a ``S.EmptySet`` instead of 
``[]``. This is because ``linsolve`` currently returns sets.

**Comment by Devyani-Divs** [comment]
@leosartaj committed that; need help writing test cases.

**Comment by leosartaj** [comment]
You should send in a PR. It will be best to discuss there.

**Comment by leosartaj** [comment]
Fixed by #9979
